substrate/client/authority-discovery/src/worker.rs

```diff
@@ -592,7 +592,7 @@ where
 					metrics.dht_event_received.with_label_values(&["value_not_found"]).inc();
 				}
 
-				if self.in_flight_lookups.remove(&hash).is_some() {
+				if self.in_flight_lookups.remove::<KademliaKey>(&hash.to_vec().into()).is_some() {
```

Comment:
Actually, the `KademliaKey` type used here is the renamed `Key` type defined in the file `rec.rs` ( now renamed `kad.rs` ).  Maybe renaming the type to something else than `KademliaKey` will make it less confusing?

Reply:
I see, keeping it called `KademliaKey` is fine. I would just import it directly from `sc_network_types`, instead of via `sc_network` -> `service` -> `types` chain.

In any case, template arguments should not be needed, because `in_flight_lookups` should have a defined key type.

Instead of using `into()` and specifying a template argument of `remove()` it is better to do it this way:
```
self.in_flight_lookups.remove(&KademliaKey::from(...))
```

Having said that, it seems `hash` and `in_flight_lookups` have the same type now, so no conversion is needed at all:
```
self.in_flight_lookups.remove(&hash)
```
(both should be `sc_network_types::kad::Key`).

Reply:
> Having said that, it seems `hash` and `in_flight_lookups` have the same type now, so no conversion is needed at all:
> 
> ```
> self.in_flight_lookups.remove(&hash)
> ```
> 
> (both should be `sc_network_types::kad::Key`).

True!! I removed all the Unnecessary conversions!